extensions/Cite/Cite.php

```diff
@@ -87,10 +87,10 @@ function wfCite( $parser ) {
 
 /**
  * @param $out OutputPage
- * @param $sk Skin
+ * @param Skin $skin
  * @return bool
  */
-function wfCiteBeforePageDisplay( $out, &$sk ) {
+function wfCiteBeforePageDisplay( OutputPage $out, Skin $skin ): bool {
```

Comment:
Hook `BeforePageDisplay` is only invoked from `OutputPage` where it receives `$this` as first parameter